### DISCUSSION TITLE
Add to documentation searching for Upsert to point to Store doc

### DISCUSSION THREAD

**Discussion by genachka** [ravendb/ravendb#14767] [General]
I tried searching for Upsert, at least on the Java docs and nothing comes up. Posted a message in Q&A https://github.com/ravendb/ravendb/discussions/14766 and promptly was pointed to using Store (thank you again @ayende for your prompt and helpful guidance). I've been using Store for inserts but hadn't realized that it can act as an Upsert as well. Perhaps improving the Store doc to make it more clear would also help, including comment on whether it replaces the entire document or is it only doing changed fields.

**Comment by ayende** [comment]
https://ravendb.net/docs/article-page/5.4/java/client-api/session/storing-entities is the relevant document.
I think this is a terminology issue. We use `store()` to refer to `create / update`.

**Reply by genachka** [reply]
Not seeing much in the way of reference that it can be used for update in that link. I think for others looking for this, would be helpful to explicitly write that it can be used for not just create but for update in the text and tag it also with Upsert so a search for update / change / upsert gets folks to this document. 

Hmm looking at the example I see a comment that I think was overlooked by me: `// send all pending operations to server, in this case only `Put` operation`
So assuming the comment and making assumptions seeing `Put` on it, I suppose that it should change only the specific fields passed. This is where I'd love for the document to make it explicitly clear a) that it can be used not just for insert but for change / update / upsert and b) that if exists with that ID that only the passed in fields will be changed so I'm not making assumptions. ;)

**Reply by ayende** [reply]
Yes, I'm not disputing the need. I'm explaining that we didn't see that because of the different terminology being used.

**Reply by ayende** [reply]
@reebhub
